src/main/java/com/craftschemers/ranchcraft/item/custom/DowsingRodItem.java

```diff
@@ -0,0 +1,34 @@
+package com.craftschemers.ranchcraft.item.custom;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemUsageContext;
+import net.minecraft.text.Text;
+import net.minecraft.util.ActionResult;
+
+public class DowsingRodItem extends Item {
+
+
+    public DowsingRodItem(Settings settings) {
+        super(settings);
+    }
+
+    @Override
+    public ActionResult useOnBlock(ItemUsageContext context) {
+
+        if (context.getWorld().isClient()) {
+
+            PlayerEntity player = context.getPlayer();
+
+            assert player != null;
```

Comment:
You shouldn't use assertions for runtime checking, they are ignored by default by the JVM. You should actually check to see if `player` is null.